plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -642,21 +648,324 @@ Invoker createInvoker() {
         return LoggingInvoker.create(isSandbox());
     }
 
+    protected static Throwable reportSuspectedMethodTooLarge(Throwable x) {
+        // Suspected groovyjarjarasm.asm.MethodTooLargeException or a
+        // org.codehaus.groovy.control.MultipleCompilationErrorsException
+        // whose collection of errors refers to MethodTooLargeException.
+        // Per review comments, we do not want to statically compile a
+        // dependency on the groovyjarjarasm.asm.MethodTooLargeException
+        // internals, so gauge hitting it via String name comparisons.
+        // Other cases may be (subclasses of) RuntimeException or Error.
+        // Note that all of MultipleCompilationErrorsException, and
+        // MethodTooLargeException and CpsCompilationErrorsException
+        // are descended from RuntimeException.
+        Throwable mtlEx = null;
+        int ecCount = 0;
+        String xStr = Functions.printThrowable(x); // includes x.getMessage() contents
+        final Pattern LINE_SEP_PATTERN = Pattern.compile("\\R");
```

Comment:
I had to look this one up.

> Any Unicode linebreak sequence, is equivalent to `\u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029]`

AFAICT
```suggestion
        final Pattern LINE_SEP_PATTERN = Pattern.compile("\r?\n");
```
suffices.